core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -0,0 +1,310 @@
+  }
+
+  @Override
+  public void resetState() {
+    mLoadPathToInfo.clear();
+  }
+
+  @Override
+  public CheckpointName getCheckpointName() {
+    return CheckpointName.LOAD_DIRECTORY;
+  }
+
+  /**
+   * Check load validation status, perform loadInfo option update accordingly.
+   * @param path the file path
+   * @param options the load options
+   * @return validation status
+   */
+  public ValidationStatus getValidationStatus(String path, LoadOptions options) {
+    if (mLoadPathToInfo.containsKey(path)) {
+      boolean isNewOption = mLoadPathToInfo.get(path).getLoadOptions().equals(options);
+      if (isNewOption) {
+        LoadInfo info = mLoadPathToInfo.get(path);
+        /*Only update bandwidth here.*/
+        long newBandWidth = options.getBandwidth();
+        info.getLoadOptions().setBandwidth(newBandWidth);
+        return ValidationStatus.Update_Options;
+      }
+      return ValidationStatus.Ignored;
+    }
+    return ValidationStatus.New;
+  }
+
+  static class Scheduler {
+    private static final int CAPACITY = 100;
+    private static final long TIMEOUT = 100;
+    private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
+    private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
+    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicLong mIdGenerator = new AtomicLong();
+
+    void schedule(Load load)
+            throws ResourceExhaustedException, InterruptedException {
+      if (mCurrentSize.get() == CAPACITY) {
+        throw new ResourceExhaustedException(
+                "Insufficient capacity to enqueue load tasks!");
+      }
+
+      boolean offered = mLoadQueue.offer(load, TIMEOUT, TimeUnit.MILLISECONDS);
+      if (offered) {
+        mCurrentSize.incrementAndGet();
+      } else {
+        LOG.warn("Cannot enqueue load to the queue, may lose track on this load!"
+                + load.getDetailedInfo());
+      }
+
+      mExecutorService.submit(() -> {
+        try {
+          runLoad(load);
+        } catch (AlluxioRuntimeException e) {
+          handleErrorOnStatuses(); // handle based on status
+        } catch (TimeoutException e) {
+          // add retry and handle timeout caused by checking available workers
+        }
+      });
+    }
+
+    void runLoad(Load load) throws AlluxioRuntimeException, TimeoutException {
+      BlockIterator<Long> blockIterator = new BlockIterator<>(load.getPath());
+
+      while (blockIterator.hasNextBatch()) {
+        // Get a worker to handle the load task.
+        ExecutionWorkerInfo worker = getNextAvailableWorker();
+        if (worker == null) { // if no workers available, continue
+          continue;
+        }
+
+        List<Long> blockIds = blockIterator.getNextBatchBlocks();
+        BlockBatch blockBatch = new BlockBatch(blockIds, getNextBatchId());
+        load.addBlockBatch(blockBatch);
+        LoadResponse status = worker.execute(blockBatch);
```

Comment:
You are running this on a single thread with a blocking API. The logic here is wrong. If you haven't thought about how to implement it, please leave it blank. Otherwise please have a working implementation.

Reply:
Discussed offline, made some changes, will have more discussions and changes for this in the future, will submit this part for now